toolchain/semantics/semantics_handle_function.cpp

```diff
@@ -6,9 +6,43 @@
 
 namespace Carbon {
 
+static auto HandleFunctionDeclarationOrDefinition(
```

Comment:
Can we call this something like "HandleFunctionSignature" or "BuildFunctionDeclaration"? The "OrDefinition" part makes me think it's actually doing definition logic.

Also, comments may be good -- in particular to emphasize that this isn't just a Handle function with an accidental `static`. :)